db/migrate/20221002120000_create_basic_tables.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class CreateBasicTables < ActiveRecord::Migration[7.0]
+  def change
+    create_table :cocktails do |t|
+      t.string :name, null: false
+      t.string :ancestry
+      t.text :description
+
+      t.timestamps
+    end
+    add_index :cocktails, :name, unique: true
+
+    create_table :glasses do |t|
+      t.string :name, null: false
+
+      t.timestamps
+    end
+    add_index :glasses, :name, unique: true
+
+    create_table :recipes do |t|
+      t.references :cocktail, foreign_key: true
+      t.references :glass, foreign_key: true
+
+      t.timestamps
+    end
```

Comment:
```suggestion
    end
    
    add_index :recipes, [: cocktail, :glass], unique: true
```

還是說這邊想要保持彈性
一種酒用同種杯子裝，有多個酒譜～？

Reply:
我覺得不排除這種可能，真的有需要可以先在 model 層擋

如果未來支援多個酒譜應該滿容易撞到這個 unique index @@
就像是 趙家的番茄炒蛋用盤子裝，鄧家的番茄炒蛋也是用盤子裝，但做的方式不同那種感覺
不同酒譜可能都是用三腳杯，但做的方式不同